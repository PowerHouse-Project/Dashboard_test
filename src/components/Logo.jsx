import "../css/logo.css";

const Logo = () => {
  const handleToggleSideBar = () => {
    document.body.classList.toggle("toggle-sidebar");
  };

  return (
    <div className="flex items-center justify-center">
      <a href="/" className="logo flex items-center">
        <span className="font-jetBrainsExtraBold text-main-light-blue-dark text-[26px]">
          PowerHouse
        </span>
      </a>
      <i
        className="bi bi-list toggle-sidebar-btn text-main-light-blue-dark"
        onClick={handleToggleSideBar}
      ></i>
    </div>
  );
};

export default Logo;
