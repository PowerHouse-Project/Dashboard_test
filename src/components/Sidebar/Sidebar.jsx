import React from "react";
import "../../css/sidebar.css";

const Sidebar = () => {
  return (
    <aside className="sidebar" id="sidebar">
      <ul className="sidebar-nav" id="sidebar-nav">
        <li className="nav-item">
          <a href="/" className="nav-link">
            <i className="bi bi-grid"></i>
            <span>Home</span>
          </a>
        </li>

        <li className="nav-item">
          <a
            href="/"
            className="nav-link collapsed"
            data-bs-target="#components-nav"
            data-bs-toggle="collapse"
          >
            <i className="bi bi-menu-button-wide"></i>
            <span>Documents</span>
          </a>
        </li>

        <li className="nav-item">
          <a
            href="/"
            className="nav-link collapsed"
            data-bs-target="#forms-nav"
            data-bs-toggle="collapse"
          >
            <i className="bi bi-journal-text"></i>
            <span>Forms</span>
          </a>
        </li>
      </ul>
    </aside>
  );
};

export default Sidebar;
