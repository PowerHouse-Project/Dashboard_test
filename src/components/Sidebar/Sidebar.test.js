import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Sidebar from "./Sidebar";
import Logo from "../Header/Logo";

describe("Sidebar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct menu items", () => {
    render(<Sidebar />);
    expect(screen.getByText("Home")).toBeInTheDocument();
    expect(screen.getByText("Devices")).toBeInTheDocument();
  });

  it("has the correct class names", () => {
    render(<Sidebar />);
    const sidebar = screen.getByRole("complementary");
    expect(sidebar).toHaveClass("sidebar");
    const navList = screen.getByRole("list");
    expect(navList).toHaveClass("sidebar-nav");
  });

  it("toggles the sidebar visibility when the toggle button is clicked", () => {
    const { container } = render(
      <>
        <Logo />
        <Sidebar />
      </>
    );

    const toggleButton = container.querySelector(".toggle-sidebar-btn");
    fireEvent.click(toggleButton);

    expect(document.body.classList.contains("toggle-sidebar")).toBe(true);

    fireEvent.click(toggleButton);

    expect(document.body.classList.contains("toggle-sidebar")).toBe(false);
  });
});
