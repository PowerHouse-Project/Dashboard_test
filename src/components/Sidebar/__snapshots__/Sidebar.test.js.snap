// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<DocumentFragment>
  <aside
    class="sidebar"
    id="sidebar"
  >
    <ul
      class="sidebar-nav"
      id="sidebar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/"
        >
          <i
            class="bi bi-grid"
          />
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link collapsed"
          data-bs-target="#components-nav"
          data-bs-toggle="collapse"
          href="/"
        >
          <i
            class="bi bi-controller"
          />
          <span>
            Devices
          </span>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link collapsed"
          data-bs-target="#components-nav"
          data-bs-toggle="collapse"
          href="/"
        >
          <i
            class="bi bi-alarm"
          />
          <span>
            Automation Schedules
          </span>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link collapsed"
          data-bs-target="#components-nav"
          data-bs-toggle="collapse"
          href="/"
        >
          <i
            class="bi bi-bar-chart"
          />
          <span>
            Energy Usage
          </span>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link collapsed"
          data-bs-target="#components-nav"
          data-bs-toggle="collapse"
          href="/"
        >
          <i
            class="bi bi-collection"
          />
          <span>
            Rooms
          </span>
        </a>
      </li>
    </ul>
  </aside>
</DocumentFragment>
`;
